src/backend/distributed/master/master_stage_protocol.c

```diff
@@ -518,6 +521,19 @@ CreateShardsOnWorkers(Oid distributedRelationId, List *shardPlacements,
 		CoordinatedTransactionUse2PC();
 	}
 
+	/* mark parallel relation accesses before opening connections */
+	if (ShouldRecordRelationAccess() && useExclusiveConnection)
+	{
+		RecordParallelDDLAccess(distributedRelationId);
+
+		/* we should mark the parent as well */
+		if (alterTableAttachPartitionCommand != NULL)
```

Comment:
@onder @marcocitus this was to check if the distributed table is a partition of another distributed table. Right ?

Reply:
Yes, this should use the function call instead.